section11/numberformat_proposed.html

```diff
@@ -123,14 +125,31 @@ <h1>Number Format Functions</h1>
       <h1>FormatNumberToString ( _intlObject_, _x_ )</h1>
 
       <p>
-        The FormatNumberToString abstract operation is called with arguments _intlObject_ (which must be an object with [[MinimumSignificantDigits]], [[MaximumSignificantDigits]], [[MinimumIntegerDigits]], [[MinimumFractionDigits]], and [[MaximumFractionDigits]] internal slots), and _x_ (which must be a Number value), and returns _x_ as a string value with digits formatted according to the five formatting parameters.
+        The FormatNumberToString abstract operation is called with arguments _intlObject_ (which must be an object with [[MinimumSignificantDigits]], [[MaximumSignificantDigits]], [[MinimumIntegerDigits]], [[MinimumFractionDigits]], and [[MaximumFractionDigits]] internal slots), and _x_ (which must be a Number value), and returns a Record containing two values: _x_ as a string value with digits formatted according to the five formatting parameters in the field [[FormattedString]], and the final floating decimal value of _x_ after rounding has been performed in the field [[RoundedNumber]].
       </p>
 
       <emu-alg>
-        1. If _intlObject_.[[MinimumSignificantDigits]] and _intlObject_.[[MaximumSignificantDigits]] are both not *undefined*, then
+        1. Let _isNegative_ be _x_ &lt; 0.
+        1. If _isNegative_, then
+          1. Let _x_ be -_x_.
+        1. If _intlObject_.[[RoundingType]] is `"significantDigits"`, then
           1. Let _result_ be ToRawPrecision(_x_, _intlObject_.[[MinimumSignificantDigits]], _intlObject_.[[MaximumSignificantDigits]]).
+        1. Else If _intlObject_.[[RoundingType]] is `"fractionDigits"`, then,
+          1. Let _result_ be ToRawFixed(_x_, _intlObject_.[[MinimumFractionDigits]], _intlObject_.[[MaximumFractionDigits]]).
         1. Else,
-          1. Let _result_ be ToRawFixed(_x_, _intlObject_.[[MinimumIntegerDigits]], _intlObject_.[[MinimumFractionDigits]], _intlObject_.[[MaximumFractionDigits]]).
+          1. Assert: _intlObject_.[[RoundingType]] is `"compactRounding"`.
+          1. Let _result_ be ToRawFixed(_x_, 0, 0).
+          1. If _result_.[[IntegerDigitsCount]] = 1, then
+            1. Let _result_ be ToRawFixed(_x_, 0, 1).
+        1. Let _minInteger_ be _intlObject_.[[MinimumIntegerDigits]].
+        1. Let _intCount_ be _result_.[[IntegerDigitsCount]].
+        1. If _intCount_ < _minInteger_, then
+          1. Let _forwardZeros_ be the String consisting of _minInteger_–_intCount_ occurrences of the character `"0"`.
+          1. Let _oldString_ be _result_.[[FormattedString]].
+          1. Let _result_.[[FormattedString]] be the concatenation of Strings _forwardZeros_ and _oldString_.
+        1. Delete _result_.[[IntegerDigitsCount]].
```

Comment:
We don't tend to use this kind of operation on records. Is there a way we could avoid introducing deletion to records? e.g., rather than doing so much imperative manipulation on this record, we could return a new one with just the two fields?

Reply:
Sounds reasonable.  Done.